kube-core/src/schema.rs

```diff
@@ -111,10 +64,91 @@ impl Visitor for StructuralSchemaRewriter {
     }
 }
 
+fn convert_to_structural(
```

Comment:
```suggestion
/// Bring all property definitions from subschemas up to the root schema,
/// since Kubernetes doesn't allow subschemas to define properties.
fn hoist_subschema_properties(
```